test/integration/short/speculative-execution-simulator-tests.js

```diff
@@ -49,9 +58,9 @@ describe('Client', function() {
             assert.strictEqual(queriedHost.nodeId, 0);
            
             // Should have only sent request to node 0.
-            assertQueryCount(0, 1);
-            assertQueryCount(1, 0);
-            assertQueryCount(2, 0);
+            assertQueryCount(node0, 1);
```

Comment:
Need to fix this code, the assertions are done in callback which is out of band.